ApollosProject/apollos-prototype
# Adds Sms Login API

## DESCRIPTION

### What does this PR do, or why is it needed?

Adds two schema methods to support SMS login.
Handles the following situations
- Logging in when you have a phone number on your account
- Creating a new user if you don't have your phone number on your account. 
- International phone numbers
- Rejecting invalid phone numbers

Does not handle the following situations
- Merging accounts when your current account doesn't have a valid phone number.

### What design trade-offs/decisions were made?
n/a

### How do I test this PR?

You should be able to test entirely in graphiql. 

Run this query first
```
mutation requestPin {
  requestSmsLoginPin(phoneNumber:"your phone number here"){
    success
  }
}
```
and then run this query

```
mutation loginWithSms {
  authenticateWithSms(phoneNumber:"your phone number", pin:"pin from sms") {
    user {
      id
      profile {
        firstName
      }
    }
    token
  }
}
```

After you through the flow once, add a phone number to your rock account and walk through the flow again. 

### What automated tests did you write?
100% coverage 💥 

## TODO:

- [x] Closes #529 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
